added web topic

HomiGrotas opened the pull request:
Hi Eldar!
Hi noticed a networks-web topic was missing, so I added one :)

Eldar1205 commented on README.md:
```diff
@@ -144,6 +144,11 @@ DISCLAIMER: All below async references were added taking into consideration only
 	* [_Pyright_](https://github.com/microsoft/pyright) is another popular tool for static type checks, more far ahead than _Mypy_ regarding support for newest Python typing features, comes built-in with VS Code Python extension and supports execution in CI technologies, however at time of writing not supported by Pycharm.
 6. <u>Exceptions</u> - [Official docs built-in exceptions reference](https://docs.python.org/3/library/exceptions.html)
 	1. Tip: Derive from _BaseException_ instead of _Exception_ in order to implement an exception type that won't be caught by general purpose _except: Exception_ blocks. This technique is used for cancellations exceptions raised by async/await libraries; general purpose exception handling shouldn't handle cancellations.
+7. ([Conributed by HomiGrotas](https://github.com/HomiGrotas)) <u>Web</u> - Interacting with web services
```
The following numbers are still 7,8, 9, etc.

HomiGrotas replied:
Yep. Which saves time increasing all following numbers

Eldar1205 replied:
I'm not sure I understand. If this is 7, the following numbers should be 8,9,... and now they are still 7,8,...

HomiGrotas replied:
Correct. However the preview seems to fix it with no need to update the numbers

Eldar1205 commented on README.md:
```diff
@@ -144,6 +144,11 @@ DISCLAIMER: All below async references were added taking into consideration only
 	* [_Pyright_](https://github.com/microsoft/pyright) is another popular tool for static type checks, more far ahead than _Mypy_ regarding support for newest Python typing features, comes built-in with VS Code Python extension and supports execution in CI technologies, however at time of writing not supported by Pycharm.
 6. <u>Exceptions</u> - [Official docs built-in exceptions reference](https://docs.python.org/3/library/exceptions.html)
 	1. Tip: Derive from _BaseException_ instead of _Exception_ in order to implement an exception type that won't be caught by general purpose _except: Exception_ blocks. This technique is used for cancellations exceptions raised by async/await libraries; general purpose exception handling shouldn't handle cancellations.
+7. ([Conributed by HomiGrotas](https://github.com/HomiGrotas)) <u>Web</u> - Interacting with web services
```
1. There's a topic `Http Client ` where `requests` module is more natural, however it's for async I/O only and for reason - blocking I/O is not something I'd recommend for backend/data engineers building data pipelines. We can list the "requests" module as well in same topic and specify explicitly it's not recommended for Production code expected to be scalable, but can be useful for other scenarios. WDYT?
2. BeautifulSoup and Selenium deserve a new topic "web scraping" which doesn't exist yet, great addition! I don't think one should be recommended over the other since they provide different capabilities, I think a quick explanation of how they differ is important for people who don't know them to know which to focus on. BTW notice the typo "Beutifulsoup" should be "BeautifulSoup".
3. Scapy and any other low-level networking aspects I'm not convinced yet are enough general purpose for backend/data engineers. It'd be interesting to open a dedicated issue on that and see if there can be a discussion, WDYT? Meanwhile I propose to add a new topic "Real time networking" or "Low level networking" (I think the latter is more accurate) and add `Scapy` there and other alternatives available.

HomiGrotas replied:
1. Sounds good
2. Ok, will do
3. Ok. It's in issue #4 and changes will be made accordingly

Eldar1205 commented on README.md:
```diff
@@ -144,6 +144,11 @@ DISCLAIMER: All below async references were added taking into consideration only
 	* [_Pyright_](https://github.com/microsoft/pyright) is another popular tool for static type checks, more far ahead than _Mypy_ regarding support for newest Python typing features, comes built-in with VS Code Python extension and supports execution in CI technologies, however at time of writing not supported by Pycharm.
 6. <u>Exceptions</u> - [Official docs built-in exceptions reference](https://docs.python.org/3/library/exceptions.html)
 	1. Tip: Derive from _BaseException_ instead of _Exception_ in order to implement an exception type that won't be caught by general purpose _except: Exception_ blocks. This technique is used for cancellations exceptions raised by async/await libraries; general purpose exception handling shouldn't handle cancellations.
+7. ([Conributed by HomiGrotas](https://github.com/HomiGrotas)) <u>Web</u> - Interacting with web services
```
typo: should be "Contributed" instead of "Conributed"

Eldar1205 reviewed (changes requested):
Thank you very much for the contribution, really appreciated!
I've reviewed the changes and provided feedback.

FYI I would like to add a more organized approach to crediting contributors so I recommend to wait with merging this PR until that point.

HomiGrotas commented:
Hi, made the following changes as I was asked:
1. Updated list numbers
2. Created web scraping topic
3. Fixed Beutifulsoup typo to BeautifulSoup
4. Added requests module to HTTP Client and warned about it's synchronous functionality

Eldar1205 commented on README.md:
```diff
@@ -144,10 +144,15 @@ DISCLAIMER: All below async references were added taking into consideration only
-9. <u>Json</u> - fast library for working with JSON, supports dataclasses serialization - [_orjson_](https://github.com/ijl/orjson)
-10. <u>Data Models</u> - represent system entities as typed data models designed for static type checks - [_pydantic_](https://pydantic-docs.helpmanual.io/):
+7.  <u>Web Scraping</u> - Using code to extract content and data from a website
+	1. Selenium - Selenium Python bindings provides a simple API to write functional and **interactive** tests using Selenium WebDriver [_Docs reference_](https://selenium-python.readthedocs.io/)
```
Can you please explain how Selenium is usable in the context of web scraping and not just testing?

Eldar1205 commented on README.md:
```diff
@@ -144,10 +144,15 @@ DISCLAIMER: All below async references were added taking into consideration only
+7.  <u>Web Scraping</u> - Using code to extract content and data from a website
+	1. Selenium - Selenium Python bindings provides a simple API to write functional and **interactive** tests using Selenium WebDriver [_Docs reference_](https://selenium-python.readthedocs.io/)
+	2. BeautifulSoup - Python library for pulling data out of HTML and XML files [_Official docs reference_](https://www.crummy.com/software/BeautifulSoup/bs4/doc/)
+	3. Scrapy - A fast high-level web crawling and web scraping framework, used to crawl websites and extract structured data from their pages. It can be used for a wide range of purposes, from data mining to monitoring and automated testing. [_Official docs reference_](https://docs.scrapy.org/)
```
Isn't it a library and not a framework? Frameworks invoke our code ([The Holywood Principle](https://dzone.com/articles/the-hollywood-principle)) while libraries are invoked by our code

HomiGrotas replied:
According to their [own description](https://pypi.org/project/Scrapy/), it's a framework

Eldar1205 replied:
Very well let's keep it, even though they don't follow the Holywood Principle so it's designed as a library

Eldar1205 reviewed (commented):
Thanks for handling previous comments!

Eldar1205 commented on README.md:
```diff
@@ -159,43 +164,44 @@ DISCLAIMER: All below async references were added taking into consideration only
-	* Recommended _httpx_ for richer feature set, for web sockets client use [_websockets_](https://websockets.readthedocs.io/en/stable/) 
-15. <u>SQL ORM</u> - object relational mapper for working with SQL databases - [_SQLAlchemy_](https://www.sqlalchemy.org/)
+	* Recommended _httpx_ for richer feature set, for web sockets client use [_websockets_](https://websockets.readthedocs.io/en/stable/)
+	3. ([Contributed by HomiGrotas](https://github.com/HomiGrotas)) Requests -  The requests module allows you to send HTTP requests using Python [_Official docs reference_](https://requests.readthedocs.io/en/latest/user/quickstart/)<br>***Note: The requests module is synchronous module and therefore it's not recommended for production code expected to be scalable, but can be useful for other scenar ...
```
Can you please move the Requests module above the recommendation to use httpx?

Also, isn't is usually called "requests" instead of "Requests"? That's how I always see it

HomiGrotas replied:
The formal name is Requests, as in the project documentation.

Eldar1205 commented on README.md:
```diff
@@ -159,43 +164,44 @@ DISCLAIMER: All below async references were added taking into consideration only
-	* Recommended _httpx_ for richer feature set, for web sockets client use [_websockets_](https://websockets.readthedocs.io/en/stable/) 
-15. <u>SQL ORM</u> - object relational mapper for working with SQL databases - [_SQLAlchemy_](https://www.sqlalchemy.org/)
+	* Recommended _httpx_ for richer feature set, for web sockets client use [_websockets_](https://websockets.readthedocs.io/en/stable/)
+	3. ([Contributed by HomiGrotas](https://github.com/HomiGrotas)) Requests -  The requests module allows you to send HTTP requests using Python [_Official docs reference_](https://requests.readthedocs.io/en/latest/user/quickstart/)<br>***Note: The requests module is synchronous module and therefore it's not recommended for production code expected to be scalable, but can be useful for other scenar ...
```
```suggestion
	3. Requests - The requests module allows you to send HTTP requests using Python [_Official docs reference_](https://requests.readthedocs.io/en/latest/user/quickstart/)<br>***Note: The requests module is synchronous module and therefore it's not recommended for production code expected to be scalable, but can be useful for other scenarios.** 
```

You may ask the all-contributors bot to add you as contributor instead :)
